### ISSUE TITLE
Feature dimension reduction: Add PCA to QueryAnalyzer (conduit)

### ISSUE THREAD

**Issue by evanvolgas** [ashita-ai/conduit#3]
## Problem

LinUCB with 387 dimensions (384 embedding + 3 metadata) requires ~4,000 samples per arm for robust convergence. In production, this means 68,000 queries before the system learns effectively.

## Analysis

**Current state:**
- Feature dim: 387
- Samples needed: 10-50× per arm = 3,870-19,350 per arm
- Total with 17 models: 65,790-328,950 queries

**With PCA (387→64):**
- Feature dim: 67 (64 embedding + 3 metadata)
- Samples needed: 640-3,350 per arm  
- Total with 17 models: 10,880-56,950 queries
- **Reduction: 75-83% fewer queries needed**

## Solution

Add optional PCA dimensionality reduction to `QueryAnalyzer` in `conduit/engines/analyzer.py`.

## Tasks

- [ ] Add PCA as optional parameter to `QueryAnalyzer.__init__()`
- [ ] Implement PCA fitting method (one-time setup)
- [ ] Add PCA transform in `analyze()` method
- [ ] Update `QueryFeatures` to handle variable embedding dims
- [ ] Add PCA model persistence (save/load trained model)
- [ ] Update `LinUCBBandit` to accept configurable `feature_dim`
- [ ] Add configuration option in `core/config.py`
- [ ] Write tests for PCA integration
- [ ] Document in README and examples

## Implementation Sketch

```python
# conduit/engines/analyzer.py
from sklearn.decomposition import PCA
import pickle

class QueryAnalyzer:
    def __init__(
        self,
        embedding_model: str = "all-MiniLM-L6-v2",
        use_pca: bool = False,
        pca_dims: int = 64,
        cache_service: CacheService | None = None
    ):
        self.embedder = SentenceTransformer(embedding_model)
        self.cache_service = cache_service
        
        # PCA for dimensionality reduction
        self.use_pca = use_pca
        self.pca: PCA | None = None
        self.pca_dims = pca_dims
        
        if use_pca:
            # Try to load pre-trained PCA
            self.pca = self._load_pca() or PCA(n_components=pca_dims)
    
    def fit_pca(self, queries: list[str], save_path: str = "models/pca.pkl") -> None:
        """Fit PCA on representative query set (one-time setup).
        
        Args:
            queries: Representative queries for PCA fitting (1000+ recommended)
            save_path: Path to save fitted PCA model
        """
        if not self.use_pca or self.pca is None:
            raise ValueError("PCA not enabled")
            
        embeddings = self.embedder.encode(queries)
        self.pca.fit(embeddings)
        
        # Save fitted model
        Path(save_path).parent.mkdir(parents=True, exist_ok=True)
        with open(save_path, "wb") as f:
            pickle.dump(self.pca, f)
            
    async def analyze(self, query: str) -> QueryFeatures:
        """Extract features from query.
        
        Returns QueryFeatures with:
        - embedding: 64 dims (if PCA) or 384 dims (if not)
        - token_count, complexity_score, domain, domain_confidence
        """
        # Get embedding
        embedding = self.embedder.encode(query)  # 384 dims
        
        # Apply PCA if enabled
        if self.use_pca and self.pca is not None:
            embedding = self.pca.transform([embedding])[0]  # 64 dims
            
        # Rest of feature extraction...
        return QueryFeatures(
            embedding=embedding.tolist(),
            token_count=len(query.split()),
            complexity_score=self._compute_complexity(query),
            domain=domain.name,
            domain_confidence=domain.confidence
        )
```

## Configuration

```python
# conduit/core/config.py
class Settings(BaseModel):
    # Query Analysis
    use_pca: bool = False
    pca_dimensions: int = 64
    pca_model_path: str = "models/pca.pkl"
    
# .env
USE_PCA=true
PCA_DIMENSIONS=64
```

## Usage

```python
# One-time PCA setup
from conduit.engines.analyzer import QueryAnalyzer

analyzer = QueryAnalyzer(use_pca=True, pca_dims=64)

# Fit on representative queries
training_queries = [...]  # 1,000+ diverse queries
analyzer.fit_pca(training_queries, save_path="models/pca.pkl")

# Production usage (loads pre-fitted PCA)
analyzer = QueryAnalyzer(use_pca=True)
features = await analyzer.analyze("What is 2+2?")
# features.embedding is now 64 dims instead of 384
```

## Testing Strategy

- Unit tests: PCA fit/transform correctness
- Integration tests: Full routing with PCA features
- Benchmark: Quality impact (expect <5% degradation)
- Performance: Faster matrix operations with smaller dims

## Expected Impact

- ✅ 75% reduction in convergence time (65K → 11K queries)
- ✅ 82% reduction in memory (smaller A matrices)
- ✅ Faster routing decisions (smaller matrix inversions)
- ✅ Minimal quality loss (<5% expected)

## References

- Analysis: Session analysis on sample requirements
- Related: conduit-benchmark issue #3 (benchmark PCA implementation)
